gobblin-runtime/src/main/java/org/apache/gobblin/runtime/GobblinMultiTaskAttempt.java

```diff
@@ -157,6 +161,11 @@ public void run()
     Pair<List<Task>, Boolean> executionResult = runWorkUnits(countDownLatch);
     this.tasks = executionResult.getFirst();
 
+    if (this.tasks.isEmpty() && this.stopped.get()) {
```

Comment:
Let me know if this understanding is correct: whenever these two conditions hold at the same time, it indicates that the cancel happens before the task is materialized and executed in the executor. ( since the regular cancel will result in tasks being non empty). Can you add a comment if this is the case, just to distinguish between the regular cancel.